### PR TITLE
Add Smoke CO Cluster to All Clusters App

### LINKED ISSUE #27239: [SMCO] SDK has to be implemented for Smoke and CO Alarm Cluster
### Summary
**All-clusters-app has to be enabled with below mentioned Attributes :**
- [ ] ExpressedState
- [ ] SmokeState
- [ ] COState
- [ ] BatteryAlert
- [ ] DeviceMuted
- [ ] TestInProgress
- [ ] HardwareFaultAlert
- [ ] EndOfServiceAlert
- [ ] InterconnectSmokeAlarm
- [ ] InterconnectCOAlarm
- [ ] ContaminationState
- [ ] SmokeSensitivityLevel
- [ ] ExpiryDate
- [ ] Global Attributes

**All-clusters-app has to be enabled with below mentioned commands :**
- [ ] SelfTestRequest

### GitHub hash of the SDK that was being used

40fb7c246671e9d37ca8bb2cfbf37f81c1509334

### Platform

raspi

### Platform Version(s)

RPI - 4, 8GB RAM

### PR DESCRIPTION
**Pull request by ericzijian1994**
Add Cluster to All Clusters App
Fixes #27239

### CODE REVIEW

File: src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp [hunk 1]
```diff
@@ -455,4 +455,9 @@ bool emberAfSmokeCoAlarmClusterSelfTestRequestCallback(CommandHandler * commandO
     return SmokeCoAlarmServer::Instance().HandleRemoteSelfTestRequest(commandObj, commandPath);
 }
 
+bool __attribute__((weak)) emberAfPluginSmokeCoAlarmSelfTestRequestCommand(EndpointId endpointId)
```
Comment (bzbarsky-apple) [inline src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:458]: Shouldn't this be left un-implemented so apps that include this cluster realize they need to implement this?

As in: this should be implemented in all-clusters-app, not here.

Comment (ericzijian1994) [reply src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp:458]: I added a `smco-stub.cpp` file to implement the `emberAfPluginSmokeCoAlarmSelfTestRequestCommand` function. Is it okay to do this?

**Comment by ericzijian1994** [comment]
@woody-apple I've rebased this branch, thanks.

**Comment by ericzijian1994** [comment]
@bzbarsky-apple @woody-apple Please review it again, thanks!

**Review by bzbarsky-apple — APPROVED** [review]

File: .github/workflows/examples-nrfconnect.yaml [hunk 2]
```diff
@@ -131,7 +131,7 @@ jobs:
                     /tmp/bloat_reports/
             - name: Build example nRF Connect SDK All Clusters App on nRF52840 DK
               run: |
-                  scripts/examples/nrfconnect_example.sh all-clusters-app nrf52840dk_nrf52840 -DCONF_FILE=prj_dfu.conf
+                  scripts/examples/nrfconnect_example.sh all-clusters-app nrf52840dk_nrf52840
```
Comment (bzbarsky-apple) [inline .github/workflows/examples-nrfconnect.yaml:134]: This does not seem related to this PR.  And this change has already happened on master...

**Review by bzbarsky-apple — APPROVED** [review]
